messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -161,8 +162,41 @@ public ActiveMQServerService(Configuration configuration, PathConfig pathConfig)
         return dataSource;
     }
 
-    Map<String, JChannel> getChannels() {
-        return channels;
+    JChannel getChannel(String name) throws Exception {
+        if (channels.containsKey(name)) {
+            return channels.get(name);
+        }
+        if (jgroupsChannels.containsValue(name)) {
+            Optional<String> groupName = jgroupsChannels.entrySet().stream().filter(entry -> name.equals(entry.getValue())).map(entry -> entry.getKey()).findFirst();
```

Comment:
Please don't use this stream stuff like this.

Every lambda costs about 1K in code space in memory, and we need to be on a constant memory diet. Plus TBH this is way harder to understand than a simple for loop.